### PR TITLE
Feature/4785 adjust consent flow to new apple api

### PR DESCRIPTION
**Pull request by freshking**
## Description

Use new EN API für pre-approving sharing of test result before adding test.

## Link to Jira

https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-4785

### CODE REVIEW

**Comment by Ein-Tim** [comment]
Hope it's ok to ask this here 🙂

This PR makes it possible to use this option, right?


Was this added with iOS 14.4?

**Comment by freshking** [comment]
> Hope it's ok to ask this here 🙂
> 
> This PR makes it possible to use this option, right?
> 

Yes

**Review by marcussc — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift [hunk 1]
```diff
@@ -338,7 +339,30 @@ class ExposureSubmissionCoordinator: NSObject, ExposureSubmissionCoordinating, R
...
+								default:
+									// present alert
+									let alert = UIAlertController.errorAlert(message: error.localizedDescription, completion: { [weak self] in
+										self?.showQRScreen(isLoading: isLoading)
+									})
+									self?.navigationController?.present(alert, animated: true, completion: nil)
+								}
+							} else {
+								// continue to scanning the qr code
+								self?.showQRScreen(isLoading: isLoading)
```
Comment (30mar) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift:359]: The comment on line 348 is a bit confusing to me maybe you can clarify, shouldn't we request Authorization  if the user is not authorized? or do we do it **after** he scans the QR code?

Comment (freshking) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift:359]: We do it after

**Review by naveeddotio — APPROVED** [review]

**Pull request merged**